src/story_gui.cpp

```diff
@@ -617,6 +619,23 @@ struct StoryBookWindow : Window {
 		this->OnInvalidateData(-1);
 	}
 
+	Point OnInitialPosition(int16_t sm_width, int16_t sm_height, [[maybe_unused]]int window_number) override
+	{
+		if (this->centered) {
+			Point pt;
+
+			int16_t default_width  = std::max(this->window_desc->GetDefaultWidth(),  sm_width);
+			int16_t default_height = std::max(this->window_desc->GetDefaultHeight(), sm_height);
+
+			pt.x = (_screen.width - default_width) / 2;
+			pt.y = (_screen.height - default_height) / 2;
+
+			return pt;
+		}
+
+		return Window::OnInitialPosition(sm_width, sm_height, window_number);
```

Comment:
I'd rather make `LocalGetWindowPlacement` non-static and add a parameter for the position which the current only caller would fill with `desc->default_pos`. Then this function can simply call `LocalGetWindowPlacement` with the appropriate `WDP` value.
That way someone looking for implementing the same functionality in another window won't be looking at duplicating this centering code.

Reply:
I tried that before this attempt, but it looks really weird. You give it a description of the window, but also a field from the description, as someone might change it. It just didn't sit right with me, and felt hackish.

But, this solution still feels hackish how ever I slice it .. so I am going back to an other way of doing this, which is just way simpler and more descriptive :D

Back&forth, back and forth .... :)